code/obj/machinery/door/airlock.dm

```diff
@@ -1610,6 +1623,21 @@ About the new airlock wires panel:
 		src.welded = null
 	src.UpdateIcon()
 
+/obj/machinery/door/airlock/proc/sword_open()
+	src.operating = -1
+	playsound(src, 'sound/machines/airlock_break_very_temp.ogg', 40, 1)
+	var/obj/item/scrap/S = new /obj/item/scrap
+	S.set_loc(src.loc)
+	play_animation("opening")
+	src.UpdateIcon(1)
+	src.set_density(0)
+	if (!istype(src, /obj/machinery/door/airlock/glass))
+		if (ignore_light_or_cam_opacity)
+			src.set_opacity(0)
+		else
+			src.RL_SetOpacity(0)
+	src.UpdateIcon()
```

Comment:
why not just call the `open` proc like emags do?

Reply:
Maybe pass a `force` param to `open` then? It's not great to duplicate the open behaviour here